tests/test_downloadermiddleware_retry.py

```diff
@@ -110,6 +110,29 @@ def test_twistederrors(self):
             == 2
         )
 
+    def test_exception_to_retry_added(self):
+        # Don't retry
+        exc = ValueError
+        req = Request("http://www.scrapytest.org/%s" % exc.__name__)
+        req = self.mw.process_exception(req, exc("foo"), self.spider)
+        self.assertEqual(req, None)
+        # Retry
+        self.mw.exceptions_to_retry = self.mw.exceptions_to_retry + (exc,)
+        req = Request("http://www.scrapytest.org/%s" % exc.__name__)
+        self._test_retry_exception(req, exc("foo"))
+
+    def test_exception_to_retry_customMiddleware(self):
+        exc = ValueError
+
+        class MyRetryMiddleware(RetryMiddleware):
+            EXCEPTIONS_TO_RETRY = RetryMiddleware.EXCEPTIONS_TO_RETRY + (exc,)
```

Comment:
As this test uses a deprecated attribute it should also check that the warning is produced.